src/core/lib/channel/channelz.cc

```diff
@@ -277,7 +278,20 @@ grpc_json* ServerNode::RenderJson() {
   return top_level_json;
 }
 
-SocketNode::SocketNode() : BaseNode(EntityType::kSocket) {}
+SocketNode::SocketNode(const char* remote_peer_string)
+    : BaseNode(EntityType::kSocket) {
+  const char* ipv6_prefix = "ipv6:";
```

Comment:
What about IPv4 addresses?

More generally, rather than doing this manually, how about using some of our existing functions for doing this parsing?  In particular, I'm thinking of `grpc_parse_uri()` and `grpc_parse_ipv{4,6}_hostport()`, all defined here:

https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/parse_address.h

This would probably require moving that parsing code from ext to lib, though.

Reply:
I could move these to lib, or I could reorganize this so that the code that created Socket is responsible for doing the splitting. I will take a shot at that since I think it is a little cleaner